Add safe navigation in case segment is nil in net http instrumentation (#3046)
newrelic/newrelic-ruby-agent: CHANGELOG.md

# New Relic Ruby Agent Release Notes

## dev

> - **Bugfix: Prevent a nil segment from causing errors in Net HTTP instrumentation**

Reviewer: ```suggestion
- **Bugfix: Prevent a `nil` segment from causing errors in Net::HTTP instrumentation**
```

>   When using JRuby, a race condition can happen that causes the segment creation to fail and returns nil. This would cause an error to occur when methods were later called on the nil segment. These methods will not be called now if it is nil, preventing that error from occurring. [PR#3046](https://github.com/newrelic/newrelic-ruby-agent/pull/3046)

Reviewer: ```suggestion
  When using JRuby, a race condition can happen that causes the segment creation to fail and return `nil`. This would cause an error to occur when methods were later called on the `nil` segment. These methods will no longer be called if the segment is `nil`, preventing that error from occurring. [PR#3046](https://github.com/newrelic/newrelic-ruby-agent/pull/3046)
```

## dev

> 

Reviewer 2: Do we normally have two lines below `##dev`?

Author: im not sure lol. it happened when i was resolving a merge conflict and i just didnt go back and remove it. I have changed it now though lol
